ldi-api/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/types/LdiInput.java

```diff
@@ -2,25 +2,43 @@
 
 import be.vlaanderen.informatievlaanderen.ldes.ldi.services.ComponentExecutor;
 
+/**
+ * Base class for the start of a LDIO workflow.
+ *
+ * Depending on its implementation, the LDI Input will be able to transfer
+ * either generated or received data
+ * onto the LDIO pipeline
+ */
 public abstract class LdiInput implements LdiComponent {
 
-	private ComponentExecutor executor;
-	private LdiAdapter adapter;
+	private final ComponentExecutor executor;
+	private final LdiAdapter adapter;
 
-	public LdiInput withExecutor(ComponentExecutor executor) {
+	/**
+	 * Creates a LdiInput with its Component Executor and LDI Adapter
+	 *
+	 * @param executor
+	 *            Instance of the Component Executor. Allows the LDI Input to pass
+	 *            data on the pipeline
+	 * @param adapter
+	 *            Instance of the LDI Adapter. Facilitates transforming the input
+	 *            data to a linked data model (RDF).
+	 */
+	public LdiInput(ComponentExecutor executor, LdiAdapter adapter) {
 		this.executor = executor;
-		return this;
-	}
-
-	public LdiInput withAdapter(LdiAdapter adapter) {
 		this.adapter = adapter;
-		return this;
 	}
 
+	/**
+	 * @return An instance of the configured Component Executor
```

Comment:
Javadoc op getters/setters vind ik dan weer overkill :p maar laat maar staan

Reply:
Totally agree. Just thought we want to be thorough for our javadocs 😄